mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/model/path/PojoModelPathPropertyNode.java

```diff
@@ -25,8 +26,17 @@ public final class PojoModelPathPropertyNode extends PojoModelPath {
 	private final String propertyName;
 
 	PojoModelPathPropertyNode(PojoModelPathValueNode parent, String propertyName) {
-		this.parent = parent;
 		Contracts.assertNotNullNorEmpty( propertyName, "propertyName" );
+		Contracts.assertTrue(
+				DOT_PATTERN.split( propertyName, -1 ).length == 1,
```

Comment:
Why split? This would work fine:

```suggestion
				!DOT_PATTERN.matcher( propertyName ).find(),
```

Reply:
honestly, I'd just go with the simple `propertyName.contains(".")`, which might even work faster than patterns. But as the task was saying:
>property name doesn’t contain any dot or any metacharacter that org.hibernate.search.mapper.pojo.model.path.PojoModelPath#parse understands

I thought I'll make it as close to the parse method as possible 😃